src/app/modules/shared_models/member_model.nim

```diff
@@ -175,6 +177,8 @@ QtObject:
       result = newQVariant(item.airdropAddress)
     of ModelRole.MembershipRequestState:
       result = newQVariant(item.membershipRequestState.int)
+    of ModelRole.UsesDefaultName:
+      result = newQVariant(resolveUsesDefaultName(item.localNickname, item.ensName, item.displayName))
```

Comment:
IMO it's preffered for the `data` call to return a member value as is and not to do any computation. The reasoning would be that `data` needs to support heavy traffic.
We've seen before some apparently harmless nim computations in `data` calls that proved to be very expensive.

This seems harmless indeed, but I don't see any benefit when compared to the approach you've used in `src/app/modules/shared_models/message_model.nim`.

Reply:
also, would be nice to have more granular `dataChanged`. We're emitting `UsesDefaultName` changed event even when it's not actually changing.

Reply:
Good idea. I updated so that the model just passes the Item's value and I also refactored a bit so that we have an Item creator function in the model to reduce duplication